test/evergreen/configure_combinations.sh

```diff
@@ -1,59 +1,84 @@
 
-options="--enable-diagnostic
---disable-shared
---disable-static --enable-python
---enable-snappy --enable-zlib --enable-lz4
---with-builtins=lz4,snappy,zlib
---enable-diagnostic --enable-python
---enable-strict --disable-shared"
+options="-DHAVE_DIAGNOSTIC=1
+-DENABLE_SHARED=0 -DENABLE_STATIC=1
+-DENABLE_STATIC=0 -DENABLE_PYTHON=1
+-DENABLE_SNAPPY=1 -DENABLE_ZLIB=1 -DENABLE_LZ4=1
+-DHAVE_BUILTIN_EXTENSION_LZ4=1 -DHAVE_BUILTIN_EXTENSION_SNAPPY=1 -DHAVE_BUILTIN_EXTENSION_ZLIB=1
+-DHAVE_DIAGNOSTIC=1-DENABLE_PYTHON=1
+-DENABLE_STRICT=1 -DENABLE_STATIC=1 -DENABLE_SHARED=0 -DWITH_PIC=1"
 
 saved_IFS=$IFS
 cr_IFS="
 "
 
 # This function may alter the current directory on failure
 BuildTest() {
-        extra_config=--enable-silent-rules
         echo "Building: $1, $2"
         rm -rf ./build || return 1
         mkdir build || return 1
         cd ./build
-        eval ../configure $extra_config "$1" "$2" \
-                 --prefix="$insdir" || return 1
-        eval make "$3" || return 1
-        make -C examples/c check VERBOSE=1 > /dev/null || return 1
-        case "$2" in
-                # Skip the install step with Python.  Even with --prefix, the
-                # install tries to write to /usr/lib64/python2.7/site-packages .
-                *enable-python* )  doinstall=false;;
-                # Non-shared doesn't yet work: library is not found at link step (??)
-                *disable-shared* ) doinstall=false;;
-                * )                doinstall=true;;
-        esac
-        if $doinstall; then
-                eval make install || return 1
-                cflags=`pkg-config wiredtiger --cflags --libs`
-                [ "$1"  == "CC=clang" ] && compiler="clang" || compiler="cc"
-                echo $compiler -o ./smoke ../examples/c/ex_smoke.c $cflags
-                $compiler -o ./smoke ../examples/c/ex_smoke.c  $cflags|| return 1
-                LD_LIBRARY_PATH=$insdir/lib ./smoke || return 1
+        eval $CMAKE_BIN $extra_config "$1" "$2" \
+                 -DCMAKE_INSTALL_PREFIX="$insdir" -G $GENERATOR ../. || return 1
+        eval $GENERATOR_CMD $PARALLEL || return 1
+        if [ "$GENERATOR" == "Unix\ Makefiles" ]; then
+            $GENERATOR_CMD -C examples/c  VERBOSE=1 > /dev/null || return 1
+        else
+            $GENERATOR_CMD examples/c/all > /dev/null || return 1
         fi
+        eval $GENERATOR_CMD install || return 1
+        (echo $2 | grep "ENABLE_SHARED=0") && wt_build="--static" || wt_build=""
+        cflags=`pkg-config wiredtiger $wt_build --cflags --libs`
+        [ "$1"  == *"clang.cmake"* ] && compiler="clang" || compiler="cc"
+        echo $compiler -o ./smoke ../examples/c/ex_smoke.c $cflags
+        $compiler -o ./smoke ../examples/c/ex_smoke.c  $cflags|| return 1
+        LD_LIBRARY_PATH="$insdir/lib:$insdir/lib64" ./smoke || return 1
```

Comment:
Why do we need `$insdir/lib64` ?

Reply:
Depending on the OS, `lib64` is also a valid install path. On RHEL machines `lib64` is the chosen install path, `lib` for Multiarch Debian/Ubuntu hosts. 
When I implemented the install target in CMake, rather than coding in fixed install paths (i.e. `lib`, `bin`), I deferred to the CMake build system to determine the correct install paths based on the host it was installing for. More specifically, we use the paths determined by `[GNUInstallDir](https://cmake.org/cmake/help/latest/module/GNUInstallDirs.html)`, which has a whole lot of custom handling based on the host its running on. Which means for RHEL targets we need to add `lib64`